widgets/numeric-refinement-list/numeric-refinement-list.js

```diff
@@ -0,0 +1,184 @@
+    RefinementList = autoHideContainer(RefinementList);
+  }
+
+  if (!container || !attributeName) {
+    throw new Error(usage);
+  }
+
+  return {
+    getConfiguration: () => {
+      return {};
+    },
+
+    render: function({results, helper, templatesConfig, state, createURL}) {
+      let templateProps = utils.prepareTemplateProps({
+        transformData,
+        defaultTemplates,
+        templatesConfig,
+        templates
+      });
+
+      let facetValues = options.map(function (option) {
+        option.isRefined = isRefined(helper.state, attributeName, option);
+        return option;
+      });
+
+      let hasNoResults = facetValues.length === 0;
+
+      let cssClasses = {
+        root: cx(bem(null), userCssClasses.root),
+        header: cx(bem('header'), userCssClasses.header),
+        body: cx(bem('body'), userCssClasses.body),
+        footer: cx(bem('footer'), userCssClasses.footer),
+        list: cx(bem('list'), userCssClasses.list),
+        link: cx(bem('link'), userCssClasses.link),
+        active: cx(bem('item', 'active'), userCssClasses.active)
+      };
+
+      ReactDOM.render(
+        <RefinementList
+          createURL={(facetValue) => createURL(refine(state, attributeName, options, facetValue))}
+          cssClasses={cssClasses}
+          facetValues={facetValues}
+          shouldAutoHideContainer={hasNoResults}
+          templateProps={templateProps}
+          toggleRefinement={toggleRefinement.bind(null, helper, attributeName, options)}
+          />,
+        containerNode
+      );
+    }
+  };
+}
+
+function isRefined(state, attributeName, option) {
+  var currentRefinements = state.getNumericRefinements(attributeName);
+
+  if (option.start !== undefined && option.end !== undefined) {
+    if (option.start === option.end) {
+      return currentRefinements['='] != undefined &&
+        currentRefinements['='].find(function (refinement) { return refinement == option.start }) !== undefined;
+    }
+  }
+
+  if (option.start !== undefined) {
+    return currentRefinements['>='] != undefined &&
+      currentRefinements['>='].find(function (refinement) { return refinement == option.start }) !== undefined;
+  }
+
+  if (option.end !== undefined) {
+    return currentRefinements['<='] != undefined &&
+      currentRefinements['<='].find(function (refinement) { return refinement == option.end }) !== undefined;
+  }
+
+  if (option.start === undefined && option.end === undefined) {
+    return Object.keys(currentRefinements).length === 0
+  }
+}
+
+function refine(state, attributeName, options, facetValue) {
+  var refinedOption = options.find(function (option) {
+    return option.name == facetValue;
+  });
+
+  var currentRefinements = state.getNumericRefinements(attributeName);
+
+  if (refinedOption.start === undefined && refinedOption.end === undefined) {
+    return state.clearRefinements(attributeName);
+  }
+
+  if (isRefined(state, attributeName, refinedOption) === false) {
```

Comment:
I would prefer having `isRefined` to always return boolean than having to test `=== false`